oss-internship-2020/libtiff/example/main_sandboxed.cc

```diff
@@ -0,0 +1,314 @@
+
+  if (!pixel_status) {
+    return absl::InternalError("unexpected pixel_status value");
+  }
+
+  SAPI_ASSIGN_OR_RETURN(
+      status_or_int,
+      api.TIFFSetFieldU1(&tif, TIFFTAG_JPEGCOLORMODE, JPEGCOLORMODE_RGB));
+  if (return_value == 0) {
+    return absl::InternalError("TIFFSetFieldU1 not available");
+  }
+
+  SAPI_ASSIGN_OR_RETURN(sz, api.TIFFTileSize(&tif));
+  if (sz != 128 * 128 * 3) {
+    return absl::InternalError(absl::StrCat("tiles are ", sz, " bytes"));
+  }
+
+  sapi::v::Array<uint8_t> buffer2_(sz);
+
+  SAPI_ASSIGN_OR_RETURN(
+      new_sz,
+      api.TIFFReadEncodedTile(&tif, kRawTileNumber, buffer2_.PtrBoth(), sz));
+  if (new_sz != sz) {
+    return absl::InternalError(absl::StrCat(
+        "Did not get expected result code from TIFFReadEncodedTile(): ", new_sz,
+        " instead of ", sz));
+  }
+
+  pixel_status = true;
+  // Checking specific pixels from the test data, 0th, 64th and 512th
+  if (status = CheckRgbPixel(0, 15, 18, 0, 0, 18, 41, buffer2_); !status.ok()) {
+    LOG(ERROR) << "CheckRgbPixel failed:\n" << status.ToString();
+  }
+  pixel_status &= status.ok();
+
+  if (status = CheckRgbPixel(64, 0, 0, 0, 0, 0, 2, buffer2_); !status.ok()) {
+    LOG(ERROR) << "CheckRgbPixel failed:\n" << status.ToString();
+  }
+  pixel_status &= status.ok();
+
+  if (status = CheckRgbPixel(512, 5, 6, 34, 36, 182, 196, buffer2_);
+      !status.ok()) {
+    LOG(ERROR) << "CheckRgbPixel failed:\n" << status.ToString();
+  }
+  pixel_status &= status.ok();
+
+  SAPI_RETURN_IF_ERROR(api.TIFFClose(&tif));
+
+  SAPI_ASSIGN_OR_RETURN(
+      status_or_tif, api.TIFFOpen(srcfile_var.PtrBefore(), r_var.PtrBefore()));
+
+  sapi::v::RemotePtr tif2(status_or_tif.value());
+  if (!tif2.GetValue()) {
+    return absl::InternalError(absl::StrCat("Could not reopen ", srcfile));
+  }
+
+  sapi::v::Array<uint32> rgba_buffer_(128 * 128);
+
+  // read as rgba
+  SAPI_ASSIGN_OR_RETURN(
+      return_value,
+      api.TIFFReadRGBATile(&tif2, 1 * 128, 2 * 128, rgba_buffer_.PtrBoth()));
+  if (return_value == 0) {
+    return absl::InternalError("TIFFReadRGBATile() returned failure code");
+  }
+
+  // Checking specific pixels from the test data, 0th, 64th and 512th
+  if (status = CheckRgbaPixel(0, 15, 18, 0, 0, 18, 41, 255, 255, rgba_buffer_);
+      !status.ok()) {
+    LOG(ERROR) << "CheckRgbaPixel failed:\n" << status.ToString();
+  }
+  pixel_status &= status.ok();
+
+  if (status = CheckRgbaPixel(64, 0, 0, 0, 0, 0, 2, 255, 255, rgba_buffer_);
+      !status.ok()) {
+    LOG(ERROR) << "CheckRgbaPixel failed:\n" << status.ToString();
+  }
+  pixel_status &= status.ok();
+
+  if (status =
+          CheckRgbaPixel(512, 5, 6, 34, 36, 182, 196, 255, 255, rgba_buffer_);
+      !status.ok()) {
+    LOG(ERROR) << "CheckRgbaPixel failed:\n" << status.ToString();
+  }
+  pixel_status &= status.ok();
+
+  SAPI_RETURN_IF_ERROR(api.TIFFClose(&tif2));
+
+  if (!pixel_status) {
+    return absl::InternalError("unexpected pixel_status value");
```

Comment:
Similar as above, mention why the pixel status check failed, the sentence itself does not hint towards what failed.